ionc/ion_reader_binary.c

```diff
@@ -368,8 +368,8 @@ iERR _ion_reader_binary_step_out(ION_READER *preader)
         ASSERT(preader->_eof ? (curr_pos == next_start) : (curr_pos <= next_start));  
         to_skip = next_start - curr_pos;
         while (to_skip > 0) {
-            if (to_skip > SIZE_MAX) {
-                one_skip = SIZE_MAX;
+            if (to_skip > MAX_SIZE) {
+                one_skip = MAX_SIZE;
```

Comment:
Hmm... based on your comment this looks like it should have been completely broken before. No test coverage?